geomstats/geometry/discrete_curves.py

```diff
@@ -572,6 +618,47 @@ def dist(self, point_a, point_b, **kwargs):
 
         return dist
 
+    # TODO (alebrigant): generalize from_vector_to_diagonal_matrix to k
+    # diagonals and modify tensorflow vstack and hstack.
+    def space_derivative(self, curve):
+        """Compute space derivative of curve using centered differences.
```

Comment:
Did you try this numpy method: https://numpy.org/doc/stable/reference/generated/numpy.gradient.html
I don't know about the other backends though

Reply:
This should work, I'll try it out.